subprojects/plugins/src/main/java/org/gradle/api/internal/plugins/GroovyJarFile.java

```diff
@@ -50,13 +50,23 @@ public boolean isGroovyAll() {
     }
 
     public boolean isIndy() {
-        return matcher.group(3) != null;
+        return "indy".equals(matcher.group(3));
+    }
+    
+    public boolean isGrooid() {
+        return "grooid".equals(matcher.group(3));
+    }
+    
+    @Nullable
+    public String getClassifier() {
+        return matcher.group(3);
```

Comment:
This does return the classifier with the leading '-' (e.g. `-grooid`). But it needs to return the pure classifier (e.g. `grooid`).